sycl/include/CL/sycl/stream.hpp

```diff
@@ -68,10 +72,25 @@ using GlobalOffsetAccessorT =
              cl::sycl::access::target::global_buffer,
              cl::sycl::access::placeholder::false_t>;
 
+inline unsigned GetFlushBufOffset(const GlobalBufAccessorT &GlobalFlushBuf,
+                                  unsigned WIOffset) {
+  return (((uint8_t)GlobalFlushBuf[WIOffset]) << 8) +
+         (uint8_t)GlobalFlushBuf[WIOffset + 1];
+}
+
+inline void SetFlushBufOffset(GlobalBufAccessorT &GlobalFlushBuf,
+                              unsigned WIOffset, unsigned Offset) {
+  GlobalFlushBuf[WIOffset] = (Offset >> 8) & 0xff;
+  GlobalFlushBuf[WIOffset + 1] = Offset & 0xff;
+}
```

Comment:
One needs a description for the arithmetic's employed in these functions.

Can we implement it more clearly? Smth like this will do:
- Store offset `static_cast<uint16_t *>(&GlobalFlushBuf[WIOffset]) = static_cast<uint16_t>(Offset);`;
- Load offset `*static_cast<uint16_t *>(&GlobalFlushBuf[WIOffset])`;

One may use `GlobalFlushBuf.get_ptr() + WIOffset` instead of `&GlobalFlushBuf[WIOffset]`

Reply:
I've tried to make this optimization but faced an error during device code compilation. Will investigate it later.
Regarding endianess it may be affected by current reading/writing option, but since we read and write in the same manner then it does not matter how it is stored in a buffer.